tests/epyccel/test_arrays.py

```diff
@@ -3998,6 +3998,17 @@ def test_numpy_arange_negative_step_2(language):
     f2 = epyccel(f1, language = language)
     np.testing.assert_array_almost_equal(f1(), f2(), decimal = 9)
 
+def test_numpy_arange_into_slice(language):
+    f1 = arrays.arr_arange_7
+    f2 = epyccel(f1, language = language)
+    n = randint(2,10)
+    m = randint(2,10)
+    x = np.array(100*np.random.random((n,m)), dtype=int)
+    x_expected = x.copy()
+    f1(x_expected)
+    f2(x)
+    np.testing.assert_array_almost_equal(x, x_expected, decimal = 9)
```

Comment:
I have just noticed that in this module we use `numpy.testing`, but according to [their own documentation](https://numpy.org/doc/stable/reference/testing.html#writing-your-own-tests) we should prefer the standard `assert` statements when using Pytest:

> pytest internally rewrites the `assert` statement to give informative output when it fails, so should be preferred over the legacy variant `numpy.testing.assert_`. Whereas plain assert statements are ignored when running Python in optimized mode with -O, this is not an issue when running tests with pytest.

Reply:
After reading more on the subject and running some tests myself, I think it is a good idea to use `numpy.testing` because it provides a quite detailed error message. Nevertheless we should only use the functions listed [here](https://numpy.org/doc/stable/reference/routines.testing.html#asserts), and avoid these [deprecated functions](https://numpy.org/doc/stable/reference/routines.testing.html#asserts-not-recommended):

- `assert_`
- `assert_almost_equal`
- `assert_approx_equal`
- `assert_array_almost_equal`
- `print_assert_equal`